### ISSUE TITLE
Fix dependency convergence

### ISSUE THREAD

**Issue by wolfc** [jboss-set/bug-clerk#87]
Enable dependency convergence enforcer rule and fix subsequent problems.

**Comment by rpelisse** [comment]
Is there any dependencies issue in 'core' ? I don't mind that dist is a mess, it's just there to build a fat jar (it should not be embarked or use by anything else than a bash script)

**Comment by rpelisse** [comment]
OK, guilty as charged, Bugclerk has some dependencies issue:

```
[WARNING] 
Dependency convergence error for org.slf4j:slf4j-api:1.7.12 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.jira:jira-rest-java-client-api:4.0.0
        +-com.atlassian.httpclient:atlassian-httpclient-api:0.23.0
          +-org.slf4j:slf4j-api:1.7.12
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.event:atlassian-event:2.3.5
        +-org.slf4j:slf4j-api:1.5.8
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-org.slf4j:slf4j-api:1.7.12
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.kie:kie-api:7.4.1.Final
    +-org.uberfire:uberfire-maven-support:1.4.0.Final
      +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.kie:kie-api:7.4.1.Final
    +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.drools:drools-core:7.4.1.Final
    +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.drools:drools-compiler:7.4.1.Final
    +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.drools:drools-decisiontables:7.4.1.Final
    +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.kie:kie-internal:7.4.1.Final
    +-org.slf4j:slf4j-api:1.7.22
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.slf4j:slf4j-simple:1.7.10
    +-org.slf4j:slf4j-api:1.7.10

[WARNING] 
Dependency convergence error for org.hamcrest:hamcrest-core:1.3 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-junit:junit:4.12
    +-org.hamcrest:hamcrest-core:1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.mockito:mockito-core:1.9.5
    +-org.hamcrest:hamcrest-core:1.1

[WARNING] 
Dependency convergence error for commons-codec:commons-codec:1.9 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.httpclient:atlassian-httpclient-plugin:0.23.0
        +-org.apache.httpcomponents:httpclient-cache:4.4.1
          +-org.apache.httpcomponents:httpclient:4.4.1
            +-commons-codec:commons-codec:1.9
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-github:aphrodite-0.7.0.Final
    +-org.kohsuke:github-api:1.77
      +-commons-codec:commons-codec:1.7
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.drools:drools-core:7.4.1.Final
    +-commons-codec:commons-codec:1.10
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.drools:drools-decisiontables:7.4.1.Final
    +-org.apache.poi:poi:3.15
      +-commons-codec:commons-codec:1.10

[WARNING] 
Dependency convergence error for com.google.guava:guava:10.0.1 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.jira:jira-rest-java-client-api:4.0.0
        +-com.google.guava:guava:10.0.1
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.event:atlassian-event:2.3.5
        +-com.google.guava:guava:11.0.2
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.google.guava:guava:10.0.1

[WARNING] 
Dependency convergence error for commons-logging:commons-logging:1.1.3 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-common:aphrodite-0.7.0.Final
    +-commons-logging:commons-logging:1.1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-domain:aphrodite-0.7.0.Final
    +-org.jboss.set:jboss-aphrodite-container:aphrodite-0.7.0.Final
      +-commons-logging:commons-logging:1.1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-domain:aphrodite-0.7.0.Final
    +-commons-logging:commons-logging:1.1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-org.springframework:spring-beans:2.5.6
        +-commons-logging:commons-logging:1.1.1
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-org.springframework:spring-beans:2.5.6
        +-org.springframework:spring-core:2.5.6
          +-commons-logging:commons-logging:1.1.1
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-commons-logging:commons-logging:1.1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-github:aphrodite-0.7.0.Final
    +-commons-logging:commons-logging:1.1.3
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-simplecontainer:aphrodite-0.7.0.Final
    +-commons-logging:commons-logging:1.1.3

[WARNING] 
Dependency convergence error for org.apache.httpcomponents:httpmime:4.4.1 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.httpclient:atlassian-httpclient-plugin:0.23.0
        +-org.apache.httpcomponents:httpmime:4.4.1
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-org.apache.httpcomponents:httpmime:4.1.2

[WARNING] 
Dependency convergence error for commons-lang:commons-lang:2.4 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.event:atlassian-event:2.3.5
        +-commons-lang:commons-lang:2.4
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-github:aphrodite-0.7.0.Final
    +-org.kohsuke:github-api:1.77
      +-commons-lang:commons-lang:2.6

[WARNING] 
Dependency convergence error for com.atlassian.util.concurrent:atlassian-util-concurrent:2.4.2 paths to dependency are:
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.jira:jira-rest-java-client-api:4.0.0
        +-com.atlassian.util.concurrent:atlassian-util-concurrent:2.4.2
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.event:atlassian-event:2.3.5
        +-com.atlassian.util.concurrent:atlassian-util-concurrent:2.4.1
and
+-org.jboss.jbossset:bugclerk-core:1.0.1.Final-SNAPSHOT
  +-org.jboss.set:jboss-aphrodite-jira:aphrodite-0.7.0.Final
    +-com.atlassian.jira:jira-rest-java-client-core:4.0.0
      +-com.atlassian.util.concurrent:atlassian-util-concurrent:2.4.2

```
However, I'm not sure exactly how to adress them...

**Comment by wolfc** [comment]
The dependency on {{org.slf4j:slf4j-simple}} is just plain wrong. It should not be exposed.
{{org.slf4j:slf4j-api}} needs to be handled through {{dependencyManagement}}.
{{org.hamcrest:hamcrest-core}} should probably be excluded from {{org.mockito:mockito-core}} to make {{junit:junit}} dictate it.
The rest is actually bad dependency convergence in {{aphrodite}}. :)

**Comment by rpelisse** [comment]
Implemented by [commit 0850980a46c18fa432b528583bcf3671b912a2cb](https://github.com/jboss-set/bug-clerk/commit/0850980a46c18fa432b528583bcf3671b912a2cb)
